apm-agent-core/src/test/java/org/example/stacktrace/ErrorCaptureTest.java

```diff
@@ -65,11 +66,37 @@ void testCulprit() {
     }
 
     @Test
-    void testUnnestNestedExceptions() {
-        final ErrorCapture errorCapture = new ErrorCapture(tracer);
-        final NestedException nestedException = new NestedException(new Exception());
-        errorCapture.setException(nestedException);
+    void testUnnestNestedException() {
+        final NestedException nestedException = new NestedException(new CustomException());
+        ErrorCapture errorCapture = tracer.captureException(nestedException, null, null);
+        assertThat(errorCapture).isNotNull();
         assertThat(errorCapture.getException()).isNotInstanceOf(NestedException.class);
+        assertThat(errorCapture.getException()).isInstanceOf(CustomException.class);
+    }
+
+    @Test
+    void testUnnestDoublyNestedException() {
+        final NestedException nestedException = new NestedException(new NestedException(new CustomException()));
+        ErrorCapture errorCapture = tracer.captureException(nestedException, null, null);
+        assertThat(errorCapture).isNotNull();
+        assertThat(errorCapture.getException()).isNotInstanceOf(NestedException.class);
+        assertThat(errorCapture.getException()).isInstanceOf(CustomException.class);
+    }
+
+    @Test
+    void testIgnoredNestedException() {
```

Comment:
another test to confirm that Nested(Custom(Nested)) also works (ie that some maintenance change doesn't make it only check the final nested exception and not the intermediate one)?